libs/libarchfpga/src/read_xml_arch_file.cpp

```diff
@@ -72,26 +72,56 @@ struct t_fc_override {
     float fc_value;
 };
 
+struct t_pin_counts {
+    int total = 0;
+    int input = 0;
+    int output = 0;
+    int clock = 0;
+
+    void sum() {
+        total = input + output + clock;
+    }
```

Comment:
I'd probably just have:
```
    void total() {
        return input + output + clock;
    }
```
which reduces the amount of state we carry around (e.g. don't need to worry about whether sum() has been called)

Reply:
Done